### PR TITLE
[WIP] Paradise Powertools

### PR DESCRIPTION
**Pull request by TheGreatNacho**
I've ported over the power tools from paradise station.
image

Power tools work like regular tools, only 4 times faster.
The Hand Drill can act as either a Screw Driver, or a Wrench depending on what setting it's on.
The Jaws of Life can act as either Wire cutters, or a crowbar depending on what setting it's on.

Sounds for these power tools have also been ported over.

Most references to tool sounds have been replaced with calling the usesound variable from the tool, and a toolspeed variable has been added to tools.

The tools require Material 2, Engineering 4 and Power 3 to be made at the protolathe.

image
_(Screenshot outdated)_
I've added in a CE toolbelt loadout. The CE will now spawn with the following:

- A hand drill
- An industrial welding tool
- The Jaws of Life
- A multitool
- A Cable coil
- A mini fire extinguisher

TODO:
- [x] Add Power research 3 to powertools.
- [ ] Add powercells for tools. Spawned tools will start with a high capacity power cell.
- [ ] Make the jaws of life about to pry open powered doors.
- [ ] Make the jaws of life be able to cut through door bolts, but destroying the door in the process.
- [x] Add toolspeeds to different welding tools.
- [ ] Add the sound of using your tool to more things that deserve it. Makes engineering more satisfying.
- [ ] Implement Paradises system of killing yourself with your tools?? (Jaws of life cutting off your head, Drill drilling through your temple)
- [ ] Remove Unathi and Tajara's ability to use power tools.

### CODE REVIEW

**Comment by Pacmandevil** [comment]
I may just be a shitcoder from above the northen border, but I think it'd be a better idea for these to be standard equipment for Engineering (which would make sense) with the normal wrench/unwrench etc times. while leaving the common supply tool significantly shittier than engineering.

The times as is are already fairly short, and making them faster would probably push things into absurdity.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -951,7 +951,7 @@ About the new airlock wires panel:
 	else if(!repairing && iscrowbar(C))
 		if(src.p_open && (operating < 0 || (!operating && welded && !src.arePowerSystemsOn() && density && (!src.locked || (stat & BROKEN)))) )
 			playsound(src.loc, 'sound/items/Crowbar.ogg', 100, 1)
-			user.visible_message("[user] removes the electronics from the airlock assembly.", "You start to remove electronics from the airlock assembly.")
+			user.visible_message("[C][user] removes the electronics from the airlock assembly.", "You start to remove electronics from the airlock assembly.")
```
Comment (Arrow768) [inline code/game/machinery/doors/airlock.dm]: Intentional ? [C][user]

**Comment by Arrow768** [comment]
There should be a downside to the electric tools -> Add a powercell and make them draw charge when they are used. So they can run out of power.

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (tedwood3) [reply code/game/machinery/doors/airlock.dm]: It's there for testing at the moment. Trying to work out why what I'm doing's not working.

**Comment by Arrow768** [comment]
I am unsure about giving the jaws of life the ability to open powered doors and cut the bolts in a door.
(Especially the bolts part)
Opening powered doors should take roughly as much time as you need to hack the door the old fashioned way. (And airlocks with security circuts installed need to be taken into account)

**Comment by tedwood3** [comment]
I can remove the feature of cutting bolts, because that is a bit overpowered.

Do you think prying open the door taking about 40 seconds would work? It takes people like Ayden Pennington 10-20 seconds to hack open a door. And if you already know the wires, it's even easier.

**Comment by Arrow768** [comment]
Yes, 40 seconds seems a good compromise.
Maybe check if the tamper proof electronics are installed and set it to 50 or 60 seconds than.

Could also a bit of rng to it.

**Comment by tedwood3** [comment]
Hopefully, I've just resolved the merge conflict. Let me know if I did it wrong.

**Comment by tedwood3** [comment]
No need to let me know. I did it wrong. Attempting to rectify.

**Comment by ProbablyMenown** [comment]
I don't understand why Tajara/Unathi would be unable to use them. They can already use normal weapons, which typically have similar triggers to powertools.

**Comment by TheGreatNacho** [comment]
Oh, that's just a joke step. I'm not actually going to do that (although I did ask the Unathi lore dev how it could work in universe)

**Comment by Arrow768** [comment]
Closing this has not been worked on for a while

**Pull request closed without merging**